src/main/java/org/jenkinsci/plugins/pipeline/utility/steps/conf/WriteYamlStep.java

```diff
@@ -121,25 +133,32 @@ public String getFunctionName() {
         }
 
         @Override
+        @Nonnull
         public String getDisplayName() {
             return "Write a yaml from an object.";
         }
     }
 
-    public static class Execution extends AbstractSynchronousNonBlockingStepExecution<Void> {
+    public static class Execution extends SynchronousNonBlockingStepExecution<Void> {
         private static final long serialVersionUID = 1L;
 
-        @StepContextParameter
         private transient TaskListener listener;
 
-        @StepContextParameter
         private transient FilePath ws;
 
         @Inject
         private transient WriteYamlStep step;
 
+        protected Execution(@Nonnull StepContext context, WriteYamlStep step) {
+            super(context);
+            this.step = step;
+        }
+
         @Override
         protected Void run () throws Exception {
+            ws = getContext().get(FilePath.class);
+            assert ws != null;
```

Comment:
Unnecessary if properly declared in required context.

Reply:
Yeah, not a huge fan of the assertions -- but I'm assuming this is a workaround for some quirk of FindBugs being grumpy?

Reply:
Yes, but findbugs doesn't know that :'(